### ISSUE TITLE
Tapping on push notification not behaving right

### ISSUE THREAD

**Issue by anserwaseem** [EnsembleUI/ensemble#1027]
When app is in **killed** state, 
1/3 times it crashes, 
and other 2/3 times, it doesn't open the specified screen i.e., just opens up Home screen

**Comment by vinothvino42** [comment]
@vusters I checked the issue and found out that getInitialMessage from FirebaseMessaging is not added. I added and tested I'm able to navigate the screen. But there are few other issues.

1. Only for this getInitialMessage, I added handleNotification method to ensemble's addCallbackAfterInitialization like this below to execute this method only after the app is initialized
```dart
Ensemble().addCallbackAfterInitialization(
          method: () => _handleNotification(isFromTerminatedState: true));
```
2. Now it's not navigating to a screen because the (Utils.globalAppKey) NavigatorState's currentContext is null, so it's failed when executing the ScreenController().navigateToScreen with the context.
3. So I added a delay (2 seconds), now the Utils.globalAppKey.currentContext is not null, so navigation is working
4. Now another issue, when the new screen is navigated before completing the home page fully it's got replaced with new widget. See the below video for more info

Link:
